### DISCUSSION TITLE
Reset specific fields for one item of useFieldArray

### DISCUSSION THREAD

**Discussion by inomn** [react-hook-form/react-hook-form#8587] [Q&A] [answered]
Hi! Editing description + codesandbox attached.

I have a simple form array with only 3 `<select>` in each form array element.
3rd field appears conditional, depending on 1st field value.
`<options>` of 2nd and 3rd field depending on 1st field value.

I want to set the 2nd and 3rd fields `value` to undefined, when the 1st field changes.

I've tried many different ways to do it (see commented code in `<Item>` component), with no success so far.
Would appreciate any help. Thanks!

Sandbox:  https://codesandbox.io/s/react-hook-form-js-forked-igzkyt

**Comment by inomn** [comment]
Up. New description, codesandbox attached.

**Comment by inomn** [comment]
Sorry for bumping this, any suggestions? :(

**Reply by bluebill1049** [reply]
I update API suitable? if you only want to reset field value: `update(0, get(defaultValues, 'yourArrayItem'))`?

**Reply by inomn** [reply]
Hi, thank you for the answer! 

> I update API suitable?

sorry, what do you mean?

> if you only want to reset field value: update(0, get(defaultValues, 'yourArrayItem'))?

Example fields I need to update:
`shipmentInformation[${index}].loadingUnit` and `shipmentInformation[${index}].unitProfile`, actually in my project it's 3 fields, in sandbox for simplification it's two.

I didn't get how to use `update(0, get(defaultValues, 'yourArrayItem'))` for my scenario, could you please take a look at sandbox https://codesandbox.io/s/react-hook-form-js-forked-igzkyt?

**Reply by inomn** [reply]
@bluebill1049 sorry for mentioning you, maybe you missed my comment

**Comment by inomn** [comment]
still can't find a solution :(

**Comment by Moshyfawn** [comment, marked as answer]
I haven't looked into your use-case yet, but I see that you're providing `undefined` `value` to the `setValue` method, which doesn't work with RHF
> value: The value for the field. This argument is required and **can not be `undefined`**.

See [Props section of RHF docs on `setValue`](https://react-hook-form.com/api/useform/setvalue)

**Reply by inomn** [reply]
Hi @Moshyfawn and thank you for pointing out that, actually it was the case. I should read the documentation more careful
